dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/createfolder/CreateFolderPresenter.java

```diff
@@ -99,33 +93,22 @@ private void initGUIElements() {
     }
 
     public void createFolder() {
-        try {
-            final String location = PathUtil.getFolderLocation(createFolderModel.getLocation(), createFolderModel
-                    .getBucketName());
-            //Instantiating create folder task
-            final CreateFolderTask createFolderTask = new CreateFolderTask(createFolderModel.getClient(),
-                    createFolderModel.getBucketName().trim(), folderNameField.textProperty().getValue().trim(),
-                    PathUtil.getDs3ObjectList(location, folderNameField.textProperty().getValue()),
-                    loggingService, resourceBundle);
-            workers.execute(createFolderTask);
-            //Handling task actions
-            createFolderTask.setOnSucceeded(event -> {
-                this.closeDialog();
-                loggingService.logMessage(folderNameField.textProperty().getValue() + StringConstants.SPACE
-                        + resourceBundle.getString("folderCreated"), LogType.SUCCESS);
-            });
-            createFolderTask.setOnCancelled(event -> this.closeDialog());
-            createFolderTask.setOnFailed(event -> {
-                this.closeDialog();
-                alert.showAlert(resourceBundle.getString("createFolderErrLogs"));
-            });
-        } catch (final Exception e) {
-            LOG.error("Failed to create folder", e);
-            loggingService.logMessage(resourceBundle.getString("createFolderErr") + StringConstants.SPACE
-                    + folderNameField.textProperty().getValue().trim() + StringConstants.SPACE
-                    + resourceBundle.getString("txtReason") + StringConstants.SPACE + e, LogType.ERROR);
+        //Instantiating create folder task
+        final CreateFolderTask createFolderTask = new CreateFolderTask(createFolderModel.getClient(),
+                createFolderModel.getBucketName().trim(), folderNameField.textProperty().getValue().trim(),
+                loggingService, resourceBundle);
+        workers.execute(createFolderTask);
+        //Handling task actions
+        createFolderTask.setOnSucceeded(event -> {
```

Comment:
Set the on succeed handler before passing the task to the execute function.  This ordering could result in a race condition.